client/src/LayoutComponents/AllStudents/AllStudents.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import searchIcon from '../../images/search.svg';
+import axios from 'axios';
+import StudentCard from '../../SharedComponents/StudentCard';
+import './AllStudents.css';
+import { Link } from 'react-router-dom';
+
+class AllStudents extends Component {
+  state = {
+    searchfield: '',
+    students: [],
+  };
+
+  componentDidMount() {
+    axios
+      .get(`/api/students`)
+      .then((res) => {
+        this.setState({
+          students: res.data,
+        });
+      })
+      .catch((err) => err);
+  }
+
+  // store the input of the search field
+  onSearchChange = (event) => {
+    this.setState({ searchfield: event.target.value });
+  };
+
+  render() {
+    const { students } = this.state;
+
+    // compare the search field with student name
+    const filterStudents = students.filter((student) => {
+      return student.name
+        .toLowerCase()
+        .includes(this.state.searchfield.toLowerCase());
+    });
+
+    // use the filtered students array to display students cards on screen
+    const allStudents = filterStudents.map((student) => {
+      return (
+        <Link to={`/student/${student.id}`} key={student.id} className="student-card-container">
+          <StudentCard studentImg={student.image} studentname={student.name} />
+        </Link>
+      );
+    });
+
+    return (
+      <div className="students-page-container">
+        <div className='search-section'>
+          <img className='search-icon' src={searchIcon} alt='' />
+          <input
+            className='search-box'
+            type='search'
+            placeholder='Search for a student...'
+            onChange={this.onSearchChange}
```

Comment:
Hey @Rahaf-96! When making forms or inputs in react, there's the concept of controlled or uncontrolled input fields.  Controlled means that react controls the input's value entirely.  In your code, you almost have a controlled input (because you have an onChange, so input changes are going into this.state) but to make it fully controlled, instead of having it a little controlled, a little uncontrolled, you also have to use your react state when rendering the input, so state changes are displayed in the input:
```
<input
  className='search-box'
  type='search'
  value={this.state.searchfield}
  and so on
```
Why? Your code works currently, but adding `value={this.state.searchfield}` will ensure that the input field is always in sync with react state if you make changes in the future, for example if you wanted to add an "X" button to clear the search field, you could do that (setting this.state.searchfield to '') and the input field would reflect that.

Reply:
Here's a link with more information:
https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/

Reply:
Thanks for the explanation!

Reply:
@tomduggan85 When I made the input field sync with the state, I couldn't type space between the first name and the last name of the student. maybe it works if I had a submit button but in my code the result of the search filtered directly without pressing on a submit button.

Reply:
Oh, sorry! I gave you bad advice about where to put the `trim()` and that's why you can't enter a space.  It might work better if you move the trim() to inside the filterStudents map like this:
```
const filterStudents = students.filter((student) => {
      return student.name
        .toLowerCase()
        .includes(this.state.searchfield.toLowerCase().trim());
    });
```
instead of having it inside onSearchChange.  Sorry about that!

Reply:
It works! thanks for the advice :))